chain-impl-mockchain/src/rewards.rs

```diff
@@ -163,6 +174,30 @@ pub fn rewards_contribution_calculation(epoch: Epoch, params: &Parameters) -> Va
     }
 }
 
+/// Added new function to ensure rewards calculation complies with research -- KH
+/// Scale the total rewards in proportion to the total delegated stake
+pub fn rewards_contribution_calculation_scaled(
+    epoch: Epoch,
+    params: &Parameters,
+    total_delegated: Stake,
+) -> Value {
+    let r = rewards_contribution_calculation(epoch, params);
+
+    // limit the rewards if necessary
+    let rl = &params.rewards_limit;
+    let res = match rl {
+        None => r,
+        Some(rlimit) => {
+            const SCALE: u128 = 10 ^ 18;
```

Comment:
this is still invalid

```suggestion
            const SCALE: u128 = 1_000_000_000_000_000_000;
```

Reply:
We have fixed this to make it consistent with your other bug changes.